browser-support: Add startswith and codepointat polyfills.

shubham-padia opened the pull request:
Fixes #8944.
~~Adds `pollyfill.js` which will contain imports to all polyfills in
order to avoid adding a new polyfill to each entry point in
webpack.assets.json . This `pollyfill.js` is placed at each entry point
in webpack.assets.json instead.~~

I implemented the code mentioned in the [comment on the original issue](https://github.com/zulip/zulip/issues/8944#issuecomment-379043205) but when writing tests for the same, it did not fulfil many of the test cases used in [a robust implentation of the same]( https://github.com/mathiasbynens/String.prototype.startsWith/blob/master/tests/tests.js)  recommended by the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith#Polyfill).

`pollyfill.js` is added to ~~each~~ entry point in the webpack(refer [here](https://babeljs.io/docs/usage/polyfill/))

I've also added an extra commit adding `string.prototype.codepointat` which was added earlier on in the project but was not added at ~~each~~ entry point. Do let me know if this wasn't needed/should be part of another PR.

timabbott commented:
I think we don't need to add this to every webpack asset; if you look at how these assets are used, I think in all (or maybe all but a couple?) code paths `common.js` is loaded on the page first.  So we should be able to just add this to `common.js` and maybe 1-2 others.  Look at the `render_bundle` calls, including the one in `base.html`.

cPhost commented:
You will also need to do `yarn install` and bump the provision version, so the new package you added will be installed otherwise `./tools/run-dev.py` will not work. And lastly it also seems like something went wrong, can you look into it - see error image below:

![screen shot 2018-04-08 at 2 00 44 pm](https://user-images.githubusercontent.com/23620441/38470792-f261d634-3b35-11e8-8c21-bad80ba35a9c.png)

shubham-padia commented:
Thanks for the review. Updated the PR. 
@timabbott  Only adding these to `common` was enough :). Also removed `polyfill.js` as imports aren't supported in all browsers atm.

@cPhost bumped up the provision version.

shubham-padia commented:
@timabbott done

timabbott commented:
Merged, thanks @shubham-padia!
